packages/caravan-clients/src/client.ts

```diff
@@ -454,6 +479,33 @@ export class BlockchainClient extends ClientBase {
     }
   }
 
+  public async getTransaction(txid: string): Promise<TransactionDetails> {
+    try {
+      let txData: any;
+
+      if (this.type === ClientType.PRIVATE) {
+        const response = await callBitcoind(
+          this.bitcoindParams.url,
+          this.bitcoindParams.auth,
+          "getrawtransaction",
+          [txid, true],
+        );
+        txData = response.result;
```

Comment:
Move this into a function in the bitcoind file (or wallet.ts? it depends on if you want this to require tx indexing to be on). Also, note that there already is a `bitcoindRawTxData` function exported from there. I think that does what you need, except it looks like it's probably broken. I think it was moved from a class and still references `this`. and that file is still js and not ts so not typed. Maybe you could clean those up and use that as a part of this PR?

Reply:
do you want me to convert the whole bitcoind.js file to ts ??